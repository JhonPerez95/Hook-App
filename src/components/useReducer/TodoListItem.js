import React from 'react';
import PropTypes from 'prop-types';

const TodoListItem = ({ item, index, handleToggle, handleDelete }) => {
  return (
    <li key={item.id} className="list-group-item list-group-item-action">
      <p
        className={`${item.done && 'complete'}`}
        onClick={() => handleToggle(item.id)}
      >
        {index + 1}. {item.description}
      </p>
      <button
        className="btn btn-danger "
        onClick={() => {
          handleDelete(item.id);
        }}
      >
        Delete
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  index: PropTypes.number.isRequired,
  item: PropTypes.object.isRequired,
  handleToggle: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
};

export default TodoListItem;
