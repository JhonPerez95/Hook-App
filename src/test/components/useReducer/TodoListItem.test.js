import React from 'react';
import { shallow } from 'enzyme';
import TodoListItem from '../../../components/useReducer/TodoListItem';
import { demoTodos } from './demoTodos';

describe('Test al component <TodoListItem/>', () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      item={demoTodos[0]}
      index={1}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );

  test('Debe mostrarse correctamente ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe llamar la funcion handleDelete  ', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalled();
  });

  test('Debe llamar la funcion handleToggle  ', () => {
    wrapper.find('p').simulate('click');
    expect(handleDelete).toHaveBeenCalled();
  });

  test('Debe mostrar el texto correctamente  ', () => {
    const descText = wrapper.find('p').text().trim();

    expect(descText).toBe(`2. ${demoTodos[0].description}`);
  });

  test('Debe tener la class Complete si done es true ', () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        item={todo}
        index={1}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );

    expect(wrapper.find('p').hasClass('complete')).toBe(true);
  });
});
