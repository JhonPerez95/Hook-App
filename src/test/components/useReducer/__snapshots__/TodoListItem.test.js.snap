// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test al component <TodoListItem/> Debe mostrarse correctamente  1`] = `
<li
  className="list-group-item list-group-item-action"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    . 
     Learning React
  </p>
  <button
    className="btn btn-danger "
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
